### PR TITLE
Adds a confirmation dialog on Exit

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Resolves https://github.com/brave/brave-browser/issues/39607

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Open Brave.
2. Go to 3 dots menu and tap Exit.
3. Make sure there is a confirmation dialog.

Screenshot_20241022_130518

### CODE REVIEW

File: browser/ui/android/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -4066,6 +4066,9 @@ If you don't accept this request, VPN will not reconnect and your internet conne
     <message name="IDS_SHOW_UNDO_WHEN_TABS_CLOSED" desc="Menu item to show undo button when tabs are closed">
       Show undo button when tabs are closed
     </message>
+    <message name="IDS_BRAVE_EXIT_CONFIRMATION" desc="A confirmation shown on Exit Brave">
+      Are you sure you want to exit Brave?
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:4070]: @rmcfadden3 @timchilds could you please confirm that it's good with such wording?

**Review by samartnik — APPROVED** [review]
lgtm

File: browser/ui/android/strings/android_brave_strings.grd (hunk 1 above)
Comment (rmcfadden3) [reply browser/ui/android/strings/android_brave_strings.grd:4070]: Seems okay to me. Maybe change the dialog header to `Exit Brave?` instead.

QQ: Is there a difference between "exit" and "close" in this case? I'm wondering if `Close Brave? | Are you sure you want to close Brave?` would be a tad more user friendly. But if `Exit` is the correct word in Android world, I can be okay with that too.

Comment (SergeyZhukovsky) [reply browser/ui/android/strings/android_brave_strings.grd:4070]: Usually on Android the confirmation dialogs titles duplicate the menu titles, like `Exit` in our case. I can definitely make it `Exit Brave`, but it's not a common practice we have with other confirmation dialogs.
@rmcfadden3 we have a menu item that is called `Exit` for a long time. I would think users expect it's Exit versus Close.
Screenshot_20241022_132300

**Comment by rmcfadden3** [comment]
@SergeyZhukovsky — ah, okay. understood. then just `Exit` is fine for the dialog header.

**Review by deeppandya — APPROVED** [review]
LGTM

**Review by AlexeyBarabash — APPROVED** [review]
++

**Pull request merged**